File: app/src/main/java/com/originb/inkwisenote/modules/smartnotes/data/AtomicNoteEntity.java
package com.originb.inkwisenote.modules.smartnotes.data;

import androidx.room.ColumnInfo;
import androidx.room.Entity;
import androidx.room.PrimaryKey;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Getter
@Setter
@NoArgsConstructor
@Entity(tableName = "atomic_note_entities")
public class AtomicNoteEntity {
    @PrimaryKey(autoGenerate = true)
    @ColumnInfo(name = "note_id")
    private long noteId;

    @ColumnInfo(name = "filename")
    private String filename;

    @ColumnInfo(name = "filepath")
    private String filepath;

    @ColumnInfo(name = "note_type")
    private String noteType; // for handwritten and text

    @ColumnInfo(name = "page_template_id")
    private long pageTemplateId;

    @ColumnInfo(name = "created_time_ms")
    private long createdTimeMillis;

    @ColumnInfo(name = "last_modified_time_ms")
    private long lastModifiedTimeMillis;
}

File: app/src/main/java/com/originb/inkwisenote2/modules/smartnotes/data/AtomicNoteEntity.kt
package com.originb.inkwisenote2.modules.smartnotes.data

import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.PrimaryKey

@Entity(tableName = "atomic_note_entities")
class AtomicNoteEntity  // No-args constructor required by Room
{
    // Getters and Setters
    @JvmField
    @PrimaryKey(autoGenerate = true)
    @ColumnInfo(name = "note_id")
    var noteId: Long = 0

    @JvmField
    @ColumnInfo(name = "filename")
    var filename: String? = null

    @JvmField
    @ColumnInfo(name = "filepath")
    var filepath: String? = null

    @JvmField
    @ColumnInfo(name = "note_type")
    var noteType: String? = null // for handwritten and text

    @JvmField
    @ColumnInfo(name = "page_template_id")
    var pageTemplateId: Long = 0

    @JvmField
    @ColumnInfo(name = "created_time_ms")
    var createdTimeMillis: Long = 0

    @JvmField
    @ColumnInfo(name = "last_modified_time_ms")
    var lastModifiedTimeMillis: Long = 0

    public override fun clone(): AtomicNoteEntity {
        val noteToSave = AtomicNoteEntity()
        noteToSave.noteId = noteId
        noteToSave.filename = filename
        noteToSave.filepath = filepath
        noteToSave.noteType = noteType
        noteToSave.pageTemplateId = pageTemplateId
        noteToSave.createdTimeMillis = createdTimeMillis
        noteToSave.lastModifiedTimeMillis = lastModifiedTimeMillis

        return noteToSave
    }
}


